barbeque/cms/toolbar.py

```diff
@@ -40,4 +42,34 @@ def populate(self):
         for menu in [menu for menu in menus if menu]:
             self.rebuild_menu(menu)
 
-toolbar_pool.register(ForceModalDialogToolbar)
+
+@toolbar_pool.register
+class SharingExtensionToolbar(ExtensionToolbar):
+    model = None
+    insert_after = None
+
+    def get_item_position(self, menu):
+        for items in menu._memo.values():
+            for item in items:
+                if str(getattr(item, 'name', None)) in (
```

Comment:
Use force_text instead of str. This library should work with Py2 and 3 (in Py2 we would need to use unicode). Please check all other usages of "str" too.